sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/json/JsonUtils.scala

```diff
@@ -48,4 +49,33 @@ object JsonUtils {
       json.sample(withReplacement = false, options.samplingRatio, 1)
     }
   }
+
+  /**
+   * Verify if the schema is supported in JSON datasource.
+   */
+  def verifySchema(schema: StructType): Unit = {
```

Comment:
The function `verifySchema` is very similar with the one in Orc/Parquet except the exception message. Should we put it into a util object?

Reply:
Hmm .. but wouldn't the supported types be very specific to data source?

Reply:
Since supported types are specific to data sources, I think we need to verify a schema in each file format implementations. But, yes.... these built-in format (orc and parquet) has the same supported types, so it might be better to move the code `veryfySchema` into somewhere (e.g., `DataSourceUtils` or something) for avoiding code duplication....